src/main/java/org/jabref/Launcher.java

```diff
@@ -11,6 +11,9 @@
 import java.util.List;
 import java.util.Map;
 
+import javax.swing.JOptionPane;
+import javax.swing.SwingUtilities;
```

Comment:
I don't think we use swing in JabRef. Can you find JavaFX equivalents and use them?

Reply:
Yes, but the javafx is not yet initialized only in JabrefGui.

Reply:
A workaround would be to add a consumer  lambda as a parameter to the method handleMultipleAppInstances  and then pass that down to JabRefGUI where you can use it to show a dialog
I did a similar thing for the FallbackExceptionHandler

Reply:
> A workaround would be to add a consumer lambda as a parameter to the method handleMultipleAppInstances and then pass that down to JabRefGUI where you can use it to show a dialog I did a similar thing for the FallbackExceptionHandler

I have tried your solution yesterday I forget to add it into origin post:
4. Nothing happened; it seems the JavaFXThread wasn't created before the instance was terminated
```
                    FallbackExceptionHandler.installExceptionHandler((exception, thread) -> {
                        UiTaskExecutor.runInJavaFXThread(() -> {
                            DialogService dialogService = Injector.instantiateModelOrService(DialogService.class);
                            dialogService.showErrorDialogAndWait("Another JabRef instance is already running. Please switch to that instance.");
                        });
                    });
```

Reply:
Creating the javafx platform is costly and takes time. So displaying a message would require to wait for the platform to be initialized.